test/test_upnp.cpp

```diff
@@ -141,7 +151,10 @@ void run_upnp_test(char const* root_filename, char const* router_model, char con
 	fclose(xml_file);
 
 	std::ofstream xml(control_name, std::ios::trunc);
-	xml.write(soap_add_response, sizeof(soap_add_response)-1);
+	if(igd_version == 1)
```

Comment:
all of these if-conditions could just be an array lookup with igd_version as the index